### PR TITLE
LIP-6: More on specifications, corrections/clarifications through the text

### PR DESCRIPTION
**Pull request by valerini**
* Adding more clarifications
* Better flow for the document
* Fix the notations (to use capital terms everywhere)
* Add a picture to lift confusion
* Adding more concrete things on specifications

### CODE REVIEW

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:36]: not in this revision, but we need to be clear that some accounts are self-managed and eventually all will be but some are also managed by the association on behalf of owners. these accounts have an auth key of all 0 and never directly submit transactions to the chain but instead are controlled via libra root ... or something like that.

Comment (valerini) [reply lips/lip-6.mdx:36]: There is a special Note section at the end of the doc on that.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:14]: Let's say LibraRoot which is a shared entity on chain between LA and LPN and use LibraRoot instead of association hereafter

File: lips/lip-6.mdx
Comment (dahliamalkhi) [reply lips/lip-6.mdx:36]: seems to be addressed in the Notes section at the bottom?

**Review by dahliamalkhi — CHANGES_REQUESTED** [review]

File: lips/lip-6.mdx
Comment (davidiw) [reply lips/lip-6.mdx:21]: or more abstract as ... is an account capable of storing a Validator Config which defines an end point capable of participating  in Libra's proof of authority consensus protocol

File: lips/lip-6.mdx
Comment (davidiw) [reply lips/lip-6.mdx:22]: we could probably refer to the network specification and leave network addresses as ambiguous: https://github.com/libra/libra/blob/master/specifications/network/network-address.md

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:23]: Validator set defines the current set of validators performing the consensus protocol. Specifically it is a vector containing validator config entries that map to the validators current states. These entries may differ...

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:24]: We should probably re-read and unify validator config and validator node where appropriate and be clear that a validator config defines either the current or potential configuration for a validator node.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:25]: maybe consider the following order:

validator node defines a participant in consensus
validator config defines identity
validator owner is responsible for this
validator operator manages this
validator set is the current participants
finally reconfiguration

File: lips/lip-6.mdx
Comment (davidiw) [reply lips/lip-6.mdx:36]: that is technically true but I prefer to see it unified so it doesn't feel so ... much as an after thought as this special note applies to /every/ validator owner in genesis.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:55]: we should evaluate if we want these fields to be lcs encoded... as of now ... they are technically in a grey area. cc: @zekun000

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:33]: I think I had some feedback here to break this up into two sections:
* a set of operations
* a set of workflows that give high level overview and dictate you through a series of operations

not sure the best ordering

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:104]: do we want to talk about the nonce scripts or admin scripts? move currently doesn't allow setting an auth key to all zeroes when creating an account, so it has to be done with the admin script

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:129]: via admin scripts this is not true, LibraRoot can as well, how are we thinking about this quirk?

Comment (davidiw) [reply lips/lip-6.mdx:129]: cc: @DavidLDill

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:130]: * Only LibraRoot or a Validator Operator of a Validator Owner which is currently in the Validator Set can trigger reconfiguration

these two points overlap a lot, curious why they are separate...

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:132]: do we want to be explicit that these apply only to move and writesets can violate these properties or should we use the term writesets? cc: @zekun000 nomenclature!

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:135]: oh, we never fixed this :O, how hard is it to turn this off? I think all uses cases in code are two distinct accounts

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:136]: this is a bit confusing as it contradicts the above... if I am an owner acting as my own operator, why can I not reconfigrue?

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:140]: I'd prefer for us to release a new lip that updates our governance policy and strike this when appropriate rather than try to accommodate versioning now.

**Review by davidiw — COMMENTED** [review]
This is a truly amazing piece of work, I know I gave a lot of feedback, but that is just indicative of how important and amazing I think this work is :D

File: lips/lip-6.mdx
Comment (zekun000) [reply lips/lip-6.mdx:132]: I don't think this is precise, multiple calls in a single txn or multiple add|remove_validator txn?

genesis is not a block, it's a single writeset txn. writeset doesn't call those functions, it's just a combined result of multiple calls, if we want to discuss nomenclature lol

Comment (davidiw) [reply lips/lip-6.mdx:132]: Great point, how does the formal verification prove this as this is a property of the executor afaiu. Should this doc even talk about blocks?

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx:135]: This is not correctly written - "If the Validator Owner did not set the Validator Operator, the Validator Owner itself acts as its own Validator Operator" - this should be here instead.

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx:136]: Only the operator can reconfigure to avoid race conditions. The Owner is its own Operator if the Owner didn't set the Operator. So if the Operator is unset, the owner = operator, so the owner can reconfigure.
Perhaps with the above correction there is no contradiction any more...

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx]: I moved the sentences about libra_root managing the validator owners up into the Workflows section.

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx]: I hope I rephrased correctly. I created a separate subsection on how the LibraRoot account is managed.
In case you would like to take a shot at re-phrasing it, feel free to do so! I'd also be happy to make changes upon your suggestions.

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx:35]: I was trying to insert an image describing the terminology, but it is not rendering correctly indeed, will try to fix!

File: lips/lip-6.mdx
Comment (dahliamalkhi) [inline lips/lip-6.mdx:107]: Once this LIP moves to Approved stage, it cannot be changed. Therefore, I don't think it is a good idea to include comments on temporary situations, like the current owners lack of key custody.

**Review by dahliamalkhi — COMMENTED** [review]
I wonder when is the TODO going to be addressed so we can move this to "Last Call" stage?

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx:107]: I don't have a strong opinion here, but I see the concern of @davidiw in that this LIP is currently the only place to describe how the system is run and if we omit it from here, then this major caveat is likely to go under the radars.

What is the procedure to update a LIP? In case e.g. the design of the validator set changes? Could these lips have versions (similar to RFC, NIST standards)?

One more question: do we plan to have a LIP on how the AOS is being run? I.e. who are the approvers, how HSMs are involved, etc.?

Comment (valerini) [reply lips/lip-6.mdx:107]: I'll talk to David Dill tomorrow, Thu, to remove the TODO.

Comment (davidiw) [reply lips/lip-6.mdx:107]: I think this might be a bit too much detail though. I think we should speak on the hand of what the abstraction does here even if the implementation isn't quite there.

I don't want to block progress on the PR, so I'll approve this PR but also leave this blurb (that can be updated in a later PR or even by me!)

LibraRoot creates good...

What is required to create an account? A public key and a name (not there)

However Owners may desire LibraRoot to maintain the accounts on their behalf ...

How are account addresses derived ...

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:109]: yeah so my reading is we should probably have 3 paragraphs

Comment (davidiw) [reply lips/lip-6.mdx:109]: Hmm.. I wanted to say how account addresses are defined is an implementation detail... but that might be too hand wavy. I want to think about this more, but this okay.

i do think this means we should have 3 paragraphs, generalized account creation process and then how it differs for the two.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:237]: @DavidLDill, when do you think this could be wrapped up?

**Review by davidiw — APPROVED** [review]

File: lips/lip-6.mdx
Comment (dahliamalkhi) [reply lips/lip-6.mdx:107]: @valerini I agree with @davidiw  's approach. we can say "owner may delegate to LibraRoot" or "LibraRoot may initially create owner account with auto delegation to LibraRoot and revoke it when Owners notify the association they have established custody of their own keys".

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx]: Done here after talking to DavidDill.

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx]: Sounds good, I tried to make it 3 paragraphs.

File: lips/lip-6.mdx
Comment (valerini) [reply lips/lip-6.mdx]: Sounds good, I tried to rephrase this large paragraph, hope it looks better now.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:33]: update this, it is blocking your pr

Comment (JoelMarcey) [reply lips/lip-6.mdx:33]: You need to add 

```
import useBaseUrl from '@docusaurus/useBaseUrl';
``` 

towards the top of the `lip-6.mdx` file, right below the YAML metadata.

File: lips/lip-6.mdx
Comment (davidiw) [inline lips/lip-6.mdx:142]: extra nl at end

**Comment by davidiw** [comment]
This is sooooo awesome. At this point, I'd like to start looking at deltas on top of this instead of just revising the core content.

**Pull request merged**
